Bluetooth: Logging: Remove use of `CONFIG_BT_DEBUG_LOG`

theob-pro opened the pull request:
See commits message.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/58800

theob-pro committed:
Bluetooth: Samples: Remove usage of `BT_DEBUG_LOG`

Remove usage of Kconfig symbol `CONFIG_BT_DEBUG_LOG` from samples. It
has been deprecated since this PR:
https://github.com/zephyrproject-rtos/zephyr/pull/56183

The Kconfig symbols has been replaced by `CONFIG_LOG=y` on most of the
cases. Or it has been removed when not needed anymore.

Signed-off-by: Théo Battrel <theo.battrel@nordicsemi.no>

theob-pro committed:
Bluetooth: Tests: Remove usage of `BT_DEBUG_LOG`

Remove usage of Kconfig symbol `CONFIG_BT_DEBUG_LOG` from tests. It has
been deprecated since this PR:
https://github.com/zephyrproject-rtos/zephyr/pull/56183

The Kconfig symbols has been replaced by `CONFIG_LOG=y` on most of the
cases. Or it has been removed when not needed anymore.

Signed-off-by: Théo Battrel <theo.battrel@nordicsemi.no>

Thalley reviewed (changes requested):
There is a reference to `CONFIG_BT_DEBUG_LOG` in https://docs.zephyrproject.org/latest/connectivity/bluetooth/bluetooth-dev.html#host-on-linux-with-an-external-controller that needs to be handled as well

theob-pro committed:
Bluetooth: Documentation: Remove reference to `BT_DEBUG_LOG`

Remove reference to deprecated Kconfig symbol `CONFIG_BT_DEBUG_LOG` in
the documentation.

Signed-off-by: Théo Battrel <theo.battrel@nordicsemi.no>
